primer.md

```diff
@@ -235,6 +236,54 @@ general, this is true for all CloudEvents attributes as well.
 This section provides additional background and design points related to some of
 the CloudEvent attributes.
 
+### String Values
+
+Some of the CloudEvent attributes are defined to be of type `String`, and
+all data types used by CloudEvent attribute are required to have a
+string-encoding. This makes for an easy serialization to even the most
+basic formats and transports that might only have a string representation of
+attributes.
+
+Even with that model, it may still be possible to use certain
+characters that could be problematic in some cases. The CloudEvents
+specification does not attempt to further restrict the valid characters used
+for attribute values so as to allow for the specification to be used in
+situations where those potentially problematic characters are desired.
+
+However, this should not be interpreted as encouragement for producers to
+push the boundaries of what characters should be used just for the sake of
+edge-testing implementations. CloudEvents is striving to help with
+interoperability while still being flexible enough to be used in unique
+situations. So, CloudEvent producers are encouraged to only use values that
+have the best chance of being properly processed by all possible formats
+and transports that their CloudEvents might be used with.
+
+### URI-reference Values
+
+Attributes of type URI-reference have quite a large set of valid values that
+can be used. For example, it is possible that the `source` attribute could be
+as short as `home` or could be more meaningful, such as
+`com.example.my-component`. The CloudEvents specification allows for a wide
```

Comment:
`example.com/my-component` sounds more like a proper URI-ref to me (unless `my-component` is a new TLD I haven't heard of yet 😉 )

Reply:
fixed :-)